libmproxy/console/common.py

```diff
@@ -161,6 +165,88 @@ def raw_format_flow(f, focus, extended, padding):
     pile.append(urwid.Columns(resp, dividechars=1))
     return urwid.Pile(pile)
 
+## common save body parts
+def save_body(path, master, state, content):
+    if not path:
+        return
+    state.last_saveload = path
+    path = os.path.expanduser(path)
+    try:
+        f = file(path, "wb")
+        f.write(str(content))
+        f.close()
+    except IOError, v:
+        master.statusbar.message(v.strerror)
+
+def ask_save_body(k, master, state, content):
+    if k == "y":
+        master.path_prompt(
+            "Save message content: ",
+            state.last_saveload,
+            save_body,
+            master,
+            state,
+            content,
+        )
+
+def which_body_save(k, master, state, flow):
+    if k == "q":
+        master.path_prompt(
+            "Save request content: ",
+            state.last_saveload,
+            save_body,
+            master,
+            state,
+            flow.request.get_decoded_content(),
+        )
+    elif k == "r":
+        if flow.response:
+            master.path_prompt(
+                "Save response content: ",
+                state.last_saveload,
+                save_body,
+                master,
+                state,
+                flow.response.get_decoded_content(),
+            )
+        else:
+            master.statusbar.message("Flow has no response")
+
+## common copy_message parts 
+def copy_message( k, master, state, message):
+    if not message:
+        # only response could be None
+        master.statusbar.message("Flow has no response")                
+        return
+
+    if pyperclip:
+        if k == "c":
+            try:
+                pyperclip.copy(message.get_decoded_content())
+            except TypeError:
```

Comment:
not right now, I have osx at home, but will try tomorrow with an ubuntu box. As far as i recall using gtk clipboard (not tried xclip) when trying to copy payload of a binary response this exception is thrown. 
I will try tomorrow with gtk and with xclip and let you know.